.env.example

```diff
@@ -2,6 +2,7 @@ APACHE_LOG_LEVEL=warn
 HTTPPORT=80
 ONI_APP_PATH=/opt/openoni
 ONI_BASE_URL=http://localhost
+#ONI_CHRONAM_API_THROTTLE=5/minute
```

Comment:
I started out with this in `settings_local(_example).py` exactly like you wrote with a chunk of comments to explain and suggest values at first while developing it, but I moved the code to `base_settings.py` and comments to the documentation after switching it to be controlled by an env var. This is both to be more 12 factor app oriented first and to not bloat the `settings_local.py` file. I think settings like this which are easily controlled with a simple bool, number, or string value shouldn't be encouraged to go in `settings_local.py` when env vars do the job. Especially in a case like this with overriding values two levels deep in a dictionary. Having people override like that is going to be more fragile. I made sure to test it both ways via env var and settings file, but want to encourage the former. This is how NCA is oriented too, right? Why diverge on this setting? It's written that way in our docs too, so I'm trying to be consistent:

> We recommend remaining configuration be set via [environment variables](https://github.com/open-oni/open-oni/blob/feature/core-api/docs/customization/configuration.md#environment-variables), but additional settings documented in [onisite/settings_base.py](https://github.com/open-oni/open-oni/blob/feature/core-api/onisite/settings_base.py) are available for complex deployments. Keep in mind that additional settings file customizations could make future Open ONI release upgrades more difficult.

And with the env var / 12 factor app support it makes sense to expose it in the Docker environment, with `.env.example` being part of that. That is dev oriented and having all the env vars in that example file is easier than devs having to go read the docs and copy all those env vars in manually to change them.

Also I think having people discover things in the settings file and then pointing back to the docs seems a little backwards. Don't we want people to read the docs first to learn what / how to configure ONI?

I considered having both the env var and leaving the override in `settings_local(_example).py` because I think it's an important capability we don't want people to miss. I think having it there highlights it more, but it'll be pretty in people's faces in the release notes and if/when we add more sales pitch-esque documentation of features it'll be highlighted there too. So I went with keeping the settings file smaller.

Reply:
Not recommending we make the file not be environment-controlled. I just didn't think it belonged in `.env.example`. Throttling doesn't make sense in a dockerized development environment - for the same reason I've argued we shouldn't have `ONI_APP_PATH` or `ONI_DB_HOST`. Using/changing these kinds of things is generally inadvisable in a development setup.

But it's sound like `.env.example` is meant as the "here's all the stuff you can configure from the environment" source; that's something I didn't fully understand. I thought it was meant solely for development and docker use. If I'm wrong, I need to file a ticket to clean it up: it shouldn't be a hidden file if people are meant to refer to it, and it needs to have some comments so prod users know which settings make sense for them vs. dev users.

Keep that in there, but let me know if I should file a new ticket.

Reply:
Yeah that makes sense from the perspective that devs wouldn't _need_ throttling for it's "prevent overuse" capability. But for manually testing it in dev environment before deploying it to production it's handy this way.

That's how I always thought of those files for dev environment - as a list of here are all the env vars if you want to test things out manually. I am more likely to be messing with the Docker stuff than most though. I assumed other devs would know not to mess with them unless they are working on Docker or testing deployment-related stuff.

So I didn't think of it as "devs will break their dev environment if they touch these, so we shouldn't put them in .env.example". Protecting devs from themselves seems like something we should design around when it makes it harder for us to work on Docker stuff. If we're worried anyone broke their dev environment, we can just tell them to copy `.env.example` back over their `.env`.

I actually do use `ONI_BASE_URL` in my dev environment because I proxy my dev apps through a local Apache instance with `/etc/hosts` domains. Then I can deny scripting to localhost URLs with NoScript so a malicious script on the web can't try to access anything I'm running locally. University Security team meeting a few weeks back went over how dev machines can be targeted this way, but I already have that vector secured :stuck_out_tongue:

Reply:
I like the idea of noting which env vars generally shouldn't be touched unless hacking on Docker infrastructure though

Reply:
Ticketed: https://github.com/open-oni/open-oni/issues/611 - resolving this one